### PR TITLE
proposal for WFCORE-5324, Ability to execute CLI script at runtime

### PR DESCRIPTION
**Pull request by jfdenise**

### CODE REVIEW

**Review by honza-kasik — COMMENTED** [review]
Thanks @jfdenise !

File: bootable-jar/WFCORE_5324_CLI_script_exec_runtime.adoc [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+* The referenced file must exist and be readable. If not, the server aborts.
+* The CLI script must be a text file (UTF-8), the file extension if present is meaningless although `.cli` extension is advised.
+* If error occurs during CLI script execution, the failing line and error are displayed, the server aborts.
+* The changes applied at boot are not persisted in the server config. A server re-started without providing the CLI script would not contain the changes
+applied by a previous execution.
+* `org.jboss.as.cli`JBoss Modules module is now provisioned whatever the set of provisioned Galleon layers. It is required to execute CLI commands.
+* Any operation that would require a server restart would make the bootable JAR server to exit, that is the nominal behavior of a bootable JAR 
+being asked to restart.
+* Some CLI commands are not exposed: `connect`, `reload`, `shutdown`, any command related to embedded server, `patch`.
+* CLI commands that can't be executed in admin-mode (eg: jdbc-driver-info) are not supported.
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply bootable-jar/WFCORE_5324_CLI_script_exec_runtime.adoc:75]: Yes, the server starts automatically in admin-mode and reloads once the script has been executed.

File: bootable-jar/WFCORE_5324_CLI_script_exec_runtime.adoc [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+* CLI commands and operations that target not provisioned subsystem would fail. For example `data-source`
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply bootable-jar/WFCORE_5324_CLI_script_exec_runtime.adoc:76]: Yes, the error shows the failing line and the reason (e.g.: command not found).

File: bootable-jar/WFCORE_5324_CLI_script_exec_runtime.adoc [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+* 
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:TBD
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply bootable-jar/WFCORE_5324_CLI_script_exec_runtime.adoc]: Fixed.

File: bootable-jar/WFCORE_5324_CLI_script_exec_runtime.adoc [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+* [x] Bootable jar
+
+== Requirements
+
+Ability to update server configuration at bootable JAR boot time by the mean of a WildFly CLI script.
+
+=== Hard Requirements
+
+* Introduce `--cli-script=<path to CLI script>`
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply bootable-jar/WFCORE_5324_CLI_script_exec_runtime.adoc]: Yes, I added a note.

File: bootable-jar/WFCORE_5324_CLI_script_exec_runtime.adoc [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+* This argument is present in the Bootable JAR help (`--help`).
+* The referenced file must exist and be readable. If not, the server aborts.
+* The CLI script must be a text file (UTF-8), the file extension if present is meaningless although `.cli` extension is advised.
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply bootable-jar/WFCORE_5324_CLI_script_exec_runtime.adoc:67]: The newLine are the one expected per operation system.Java handles it: "A line is considered to be terminated by any one of a line feed ('\n'), a carriage return ('\r'), or a carriage return followed immediately by a linefeed."

**Review by jmesnil — APPROVED** [review]

**Comment by jmesnil** [comment]
@honza-kasik sorry, I may have merged the PR a bit too fast. I should have wait for your approval (even though the feature PR has already been merged and released with WildFly Core 15.0.0.Final)

**Comment by honza-kasik** [comment]
@jmesnil No problem! The only thing that is missing is "testing by" chapter which can be filled after RFE receives rank. However I believe it will be tested by QE most probably.

**Pull request merged**